src/main/java/org/lukhnos/nnio/file/FileVisitResult.java

```diff
@@ -20,5 +20,8 @@
  * Substitute for {@link java.nio.file.FileVisitResult}.
  */
 public enum FileVisitResult {
-  CONTINUE
+  CONTINUE,
+  SKIP_SIBLINGS,
```

Comment:
Does your code (or the code you intend to work on) use these enums? If so, how much trouble would it be to implement the corresponding behavior in `FIles.java`? It seems weird to me that this PR adds new enums without honoring their semantics…

Reply:
Per spec continue does what it should, it continues.
Terminate is very difficult, since how would you know when to terminate recursion, spec only says terminate, and nothing about returned path. (It should terminate on preVisit and postVisit as I can see). What would you suggest for signalling a terminate down the call-chain?
SKIP_SUBTREE and SKIP_SIBLINGS are only relevant in the result of preVisit, SKIP_SUBTREE just calls postVisit and returns and SKIP_SIBLINGS is also difficult to signal, since it must execute a break in the loop below.
What would your idea be for signalling, since we can't really modify the method signature. (Calling our own recursive function and passing around an object would be possible, but do you have any other ideas?)

Reply:
Good questions. Let's take a step back though. What is the call site (code in Next cloud) like? Does it use only CONTINUE or does it also use the other enums? If only CONTINUE is used, it seems that the PR's improvement is handling the null pointer, similar to your previous PR. If that's the case, how about removing the unused enums? I'm comfortable with taking the changes in Files.java only.

Reply:
Nextcloud only uses CONTINUE and null for filtering specific directories based on their path, but null should probably just be SKIP_SUBTREE. But I would be fine with only keeping the CONTINUE.

Reply:
Interesting. According to https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#walkFileTree-java.nio.file.Path-java.util.Set-int-java.nio.file.FileVisitor- , if a visitor returns null, an NPE will be thrown. It seems nnio diverged from the spec and now NextCloud is depending on this divergence.

I leave it to you to decide whether you want to make the broader changes (i.e. introduce SKIP_SUBTREE, throw NPE upon null, then make the corresponding changes in Next could). If you decide to keep things simple, then feel free to revert the additional enums, and this PR will be good to go.

Reply:
I would keep SKIP_SUBTREE, since that is implicitly impemented (does not visit further if it did not get continue), but the null check could introduce issues anywhere else. Would you be ok with that?

Reply:
Would you mind adding a comment near your changes in Files.java that SKIP_SUBTREE is implicitly handled as a result? Otherwise your proposal looks good to me.

Reply:
Ok, did it. Is the comment too long though?

Reply:
Don't worry about it. Thanks for the PR!